### PR TITLE
Counter

### PR DESCRIPTION
**Pull request by GoralTomaszGorski**
Hello Marcin

I prepared visits counter funtion.
Plese chack if it acceptable it will be great for me.

### CODE REVIEW

File: src/main/java/goral/psychotherapistoffice/web/AboutMeController.java [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
+
+    public AboutMeController(CounterService counterService) {
+        this.counterService = counterService;
+    }
+
+
     @GetMapping("/omnie")
-    public String therms(Model model){
+    public String therms(Model model, HttpServletRequest request){
+        counterService.httpParameterToIncrement(request);
```
Comment (gorskimarcin96) [inline src/main/java/goral/psychotherapistoffice/web/AboutMeController.java:22]: you added `counterService.httpParameterToIncremen` to all controllers. when i had small experience i make same

you can try create one for them :)

https://www.baeldung.com/spring-mvc-handlerinterceptor#Interceptor
```java
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import org.springframework.web.servlet.HandlerInterceptor;
import org.springframework.web.servlet.ModelAndView;

public class PageVisitInterceptor implements HandlerInterceptor {

    private final CounterService counterService;

    public AboutMeController(CounterService counterService) {
        this.counterService = counterService;
    }

    @Override
    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
            throws Exception {
        return true;
    }

    @Override
    public void postHandle(HttpServletRequest request, HttpServletResponse response, Object handler,
            ModelAndView modelAndView) throws Exception {
        counterService.httpParameterToIncrement(request);
    }

    @Override
    public void afterCompletion(HttpServletRequest request, HttpServletResponse response, Object handler,
            Exception ex) throws Exception {
       //empty function
    }
}
```

File: src/main/java/goral/psychotherapistoffice/domain/counter/Counter.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    @Column(name = "entry", nullable = false)
+    private int entry;
+
+    @Column(name = "date", nullable = false)
+    private Date date;
+
+    @Column(name = "url", nullable = false)
+    private String url;
+
```
Comment (gorskimarcin96) [inline src/main/java/goral/psychotherapistoffice/domain/counter/Counter.java:32]: we can add optional column user_id. what do you think?

Comment (gorskimarcin96) [inline src/main/java/goral/psychotherapistoffice/domain/counter/Counter.java:30]: we can add optional column user_id. what do you think?

File: src/main/java/goral/psychotherapistoffice/web/AboutMeController.java (hunk 1 above)
Comment (gorskimarcin96) [inline src/main/java/goral/psychotherapistoffice/web/AboutMeController.java:22]: you added `counterService.httpParameterToIncremen` to all controllers. when i had small experience i make same

you can try create one for them :)

https://www.baeldung.com/spring-mvc-handlerinterceptor#Interceptor
```java
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import org.springframework.web.servlet.HandlerInterceptor;
import org.springframework.web.servlet.ModelAndView;

public class PageVisitInterceptor implements HandlerInterceptor {

    private final CounterService counterService;

    public PageVisitInterceptor(CounterService counterService) {
        this.counterService = counterService;
    }

    @Override
    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
            throws Exception {
        return true;
    }

    @Override
    public void postHandle(HttpServletRequest request, HttpServletResponse response, Object handler,
            ModelAndView modelAndView) throws Exception {
        counterService.httpParameterToIncrement(request);
    }

    @Override
    public void afterCompletion(HttpServletRequest request, HttpServletResponse response, Object handler,
            Exception ex) throws Exception {
       //empty function
    }
}
```

**Review by gorskimarcin96 — APPROVED** [review]

File: src/main/java/goral/psychotherapistoffice/web/AboutMeController.java (hunk 1 above)
Comment (GoralTomaszGorski) [reply src/main/java/goral/psychotherapistoffice/web/AboutMeController.java:22]: my first thought was to prepared one controller for all class
but it didn't work
@Controller
public class StatisticsController {

    @Autowired
    private CounterService counterService;

    @GetMapping("/statistics")
    public String showStatistics(Model model) {

        Map<String, Integer> urlVisits = new HashMap<>();
        urlVisits.put("Strona główna", counterService.getVisitsForUrl("http://localhost:8080/"));
        urlVisits.put("Strona użytkownika", counterService.getVisitsForUrl("http://localhost:8080/user"));
        urlVisits.put("O nas", counterService.getVisitsForUrl("http://localhost:8080/about"));
        urlVisits.put("Oferta", counterService.getVisitsForUrl("http://localhost:8080/offer"));

        Map<String, Integer> urlRefreshes = new HashMap<>();
        urlRefreshes.put("Strona główna", counterService.getRefreshesForUrl("http://localhost:8080/"));
        urlRefreshes.put("Strona użytkownika", counterService.getRefreshesForUrl("http://localhost:8080/user"));
        urlRefreshes.put("O nas", counterService.getRefreshesForUrl("http://localhost:8080/about"));
        urlRefreshes.put("Oferta", counterService.getRefreshesForUrl("http://localhost:8080/offer"));


        model.addAttribute("urlVisits", urlVisits);
        model.addAttribute("urlRefreshes", urlRefreshes);

        // Zwróć nazwę widoku Thymeleaf
        return "statistics";
    }

Comment (GoralTomaszGorski) [reply src/main/java/goral/psychotherapistoffice/web/AboutMeController.java:22]: I will tried improve code the in the way as you suggest today.
Thank you Marcin:)

**Pull request merged**